tests/unit/single_table/test_base.py

```diff
@@ -561,3 +563,990 @@ def test_update_transformers(self):
         assert len(field_transformers) == 2
         assert isinstance(field_transformers['col1'], GaussianNormalizer)
         assert isinstance(field_transformers['col2'], GaussianNormalizer)
+
+    @patch('sdv.single_table.base.DataProcessor')
+    def test__set_random_state(self, mock_data_processor):
+        """Test that error is raised when this is not implemented."""
+        # Setup
+        rng_seed = Mock()
+        metadata = Mock()
+        instance = BaseSynthesizer(metadata)
+        instance._model = Mock()
+
+        # Run
+        instance._set_random_state(rng_seed)
+
+        # Assert
+        instance._model.set_random_state.assert_called_once_with(rng_seed)
+
+    def test__filter_conditions(self):
```

Comment:
Maybe add a few more `'Software Engineer'` columns and `80.` columns, to make sure it's correctly conditioning on both of them (rn if it only conditions on one of them, the result would be the same)